week9/avenger.py

```diff
@@ -0,0 +1,126 @@
+import datetime
+
+
+class Avenger:
+    def __init__(self, record):
+        """
+        Initializes the object with a dictionary-based record.
+        Args:
+            record (dict): Dictionary-based record of Avenger data
+        """
+        self.data = record
+        self.assignedURL = self.data['url']
+        self.name = self.data['name_alias']
+        self.appearancesInComics = int(self.data['appearances'])
+        self.current_status = self.bool_parser(self.data['current'])
+        self.avengerGender = self.data['gender']
+        self.notesData = (self.data['notes'].strip('\n'))
+        self.joinYear = int(self.data['year'])
+        self.dateJoined = datetime.date(int(self.data['year']), self.getmonth(self.data['full_reserve_avengers_intro']), 1)
```

Comment:
you can split on the comma here. It's not necessary, it will just wrap and still be executable.